src/DevBetterWeb.Web/Pages/User/MyProfile/Billing.cshtml.cs

```diff
@@ -0,0 +1,81 @@
+﻿using DevBetterWeb.Core.Interfaces;
+using DevBetterWeb.Core.Specs;
+using Microsoft.AspNetCore.Identity;
+using Microsoft.AspNetCore.Mvc.RazorPages;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Authorization;
+using DevBetterWeb.Core;
+using DevBetterWeb.Infrastructure.Identity.Data;
+
+namespace DevBetterWeb.Web.Pages.User.MyProfile
+{
+  [Authorize(Roles = AuthConstants.Roles.ADMINISTRATORS)]
+  public class BillingModel : PageModel
+  {
+#nullable disable
+    public UserBillingViewModel UserBillingViewModel { get; private set; }
+#nullable enable
+    private readonly IRepository _repository;
+    private readonly UserManager<ApplicationUser> _userManager;
+    private readonly IMemberRegistrationService _memberRegistrationService;
+
+    public BillingModel(IRepository repository, 
+      UserManager<ApplicationUser> userManager,
+      IMemberRegistrationService memberRegistrationService)
+    {
+      _repository = repository;
+      _userManager = userManager;
+      _memberRegistrationService = memberRegistrationService;
+    }
+
+    public async Task OnGetAsync()
+    {
+      var currentUserName = User.Identity!.Name;
+      var applicationUser = await _userManager.FindByNameAsync(currentUserName);
+
+      var spec = new MemberByUserIdWithBillingActivitiesSpec(applicationUser.Id);
+      var member = await _repository.GetAsync(spec);
+
+      if (member == null)
+      {
+        member = await _memberRegistrationService.RegisterMemberAsync(applicationUser.Id);
+      }
+
+      UserBillingViewModel = new UserBillingViewModel(member);
+    }
+
+    public string GetOverviewBody()
```

Comment:
Building these messages could be done somewhere in Core rather than in UI layer.